core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -403,6 +391,16 @@ public long spill(long numBytes) throws IOException {
     public void remove() {
       throw new UnsupportedOperationException();
     }
+
+    private void handleFailedDelete() {
+      // remove the spill file from disk
+      File file = spillWriters.removeFirst().getFile();
+      if (file != null && file.exists()) {
+        if (!file.delete()) {
```

Comment:
The two if conditions could be collapsed

Reply:
Right, will collapse those :)

Reply:
fixed